tsl/src/hypercore/hypercore_handler.c

```diff
@@ -279,14 +279,15 @@ lazy_build_hypercore_info_cache(Relation rel, bool create_chunk_constraints,
 HypercoreInfo *
 RelationGetHypercoreInfo(Relation rel)
 {
-	if (NULL == rel->rd_amcache)
-		rel->rd_amcache = lazy_build_hypercore_info_cache(rel,
-														  true /* create constraints */,
-														  NULL /* compressed rel created */);
+	/*coverity[tainted_data_downcast : FALSE]*/
+	HypercoreInfo *info = rel->rd_amcache;
```

Comment:
Do you have a link to this issue in the Coverity report? I can't find it. Just curious, this looks so weird, why did it decide to mark this pointer cast specifically...

Reply:
```suggestion
	/*coverity[tainted_data_downcast : FALSE]*/
	HypercoreInfo *info = (HypercoreInfo *)rel->rd_amcache;
```
Shouldn't we do explicit cast here??

Reply:
> Shouldn't we do explicit cast here??

I can add an explicit cast if you feel strongly about it but cast to and from `void*` is well-defined in C and adding an explicit cast does not change anything. There are linters that warn about this though, so that might be good enough a reason to have a cast. It's more a matter of what coding guidelines we have, or want to have.